overdue/src/test/java/org/killbill/billing/overdue/config/TestOverdueConfig.java

```diff
@@ -75,4 +80,33 @@ public void testParseConfig() throws Exception {
 
         Assert.assertEquals(c.getOverdueStatesAccount().getFirstState().getName(), "OD1");
     }
+
+    @Test(groups = "fast", description = "https://github.com/killbill/killbill/issues/1497")
+    public void testBadOverdueXml() throws Exception {
+        final OverdueConfig overdueConfig = getOverdueConfig("BadOverdue.xml");
+
+        final OverdueStatesAccount statesAccount = overdueConfig.getOverdueStatesAccount();
+        Assert.assertNotNull(statesAccount);
+
+        final Period reevaluationInterval = statesAccount.getInitialReevaluationInterval();
+        Assert.assertNull(reevaluationInterval);
+
+        final OverdueState[] overdueStates = statesAccount.getStates();
+        Assert.assertNotNull(overdueStates);
+        Assert.assertEquals(overdueStates.length, 4);
+
+        for (final OverdueState state : overdueStates) {
+            Assert.assertNotNull(state.getAutoReevaluationInterval());
+
+            final OverdueCondition condition = state.getOverdueCondition();
+            Assert.assertNotNull(condition);
+
+            Assert.assertNotNull(condition.getTotalUnpaidInvoiceBalanceEqualsOrExceeds());
+
+            final Duration duration = condition.getTimeSinceEarliestUnpaidInvoiceEqualsOrExceeds();
+            Assert.assertNull(duration);
+
+            // Thus, calling method in duration will throw NullPointerException
+        }
+    }
 }
```

Comment:
I'm thinking whether it would be good to add some more tests for other overdue configuration files that have some of the other conditions like `timeSinceEarliestUnpaidInvoiceEqualsOrExceeds`, just to confirm that this fix has not caused any regression issues? @sbrossie, what do you think?